.github/ISSUE_TEMPLATE/release.md

```diff
@@ -0,0 +1,31 @@
+---
+name: Release
+about: Announce and track an upcoming release
+labels:
+  - release
+---
+
+|  |  |
+| --- | --- |
+| Release version | YYYY.MM.X |
+| Planned release date | YYYY-MM-DD |
+| Status | On-track |
+| Release manager | @github_handle |
```

Comment:
Wonder if this is redundant for us given it is often the same person and very often the person who raised the issue

Reply:
Yeah, that's a good point that this will often be redundant. I have a slight preference for being explicit here and including this line. I don't think it's a big deal either way though

Reply:
It would be a little odd if someone assigned someone else as release manager here. That's the only reason this comes to mind. Though no strong feelings

Reply:
Yeah, agreed. I think explicitly naming the release manager here has a small amount of value for someone who is less familiar with our release process. If, for example, a maintainer of some project that uses Dask and has an upstream build identifies a large breaking change then they would see the GitHub handle for the person who is the release manager and could ping them directly. Again, probably not a huge deal either way

Let's go ahead and keep this line for now, try it out, and if we find it cumbersome then we can remove it